setup.py

```diff
@@ -640,15 +672,21 @@ def find_requirements_for_alias(alias_to_look_for: Tuple[str, str]) -> List[str]
         raise Exception(f"The extra {new_extra} is missing for alias {deprecated_extra}")
 
 
-# Add extras for all deprecated aliases. Requirements for those deprecated aliases are the same
-# as the extras they are replaced with
-for alias, extra in EXTRAS_DEPRECATED_ALIASES.items():
-    requirements = EXTRAS_REQUIREMENTS.get(extra) if extra != '' else []
-    if requirements is None:
-        raise Exception(f"The extra {extra} is missing for deprecated alias {alias}")
-    # Note the requirements are not copies - those are the same lists as for the new extras. This is intended.
-    # Thanks to that if the original extras are later extended with providers, aliases are extended as well.
-    EXTRAS_REQUIREMENTS[alias] = requirements
+def add_extras_for_all_deprecated_aliases() -> None:
+    """
+    Add extras for all deprecated aliases. Requirements for those deprecated aliases are the same
+    as the extras they are replaced with.
+    The requirements are not copies - those are the same lists as for the new extras. This is intended.
+    Thanks to that if the original extras are later extended with providers, aliases are extended as well.
+    """
+    for alias, extra in EXTRAS_DEPRECATED_ALIASES.items():
+        requirements = EXTRAS_REQUIREMENTS.get(extra) if extra != '' else []
+        if requirements is None:
+            raise Exception(f"The extra {extra} is missing for deprecated alias {alias}")
+        EXTRAS_REQUIREMENTS[alias] = requirements
+
+
+add_extras_for_all_deprecated_aliases()
```

Comment:
Do these need to be called at the top level, or could/should the be called in the "if __main__" block instead?

Reply:
The only reason I made it here is to be able to remove the whole section for "deprecated aliases. But I think I can move them to main indeed. In pre-commit where I check whether setup.py and documentation are synchronized, I already call the one function which I call in main. There might be some dependencies between the sequence of calls and definitions of dicts/arrays, but I think I can solve all of them. Let me see.

Reply:
Just thinking there are some cases where we `import setup` and might not want these called.

But if that isn't the case and everywhere we import it we'd want these called then we can leave this as it is.

Reply:
I looked at it and yeah, we pretty much always want them called. The main reason I added the methods is to make them named/pydoced and also it makes it better to reuse some of the common variable names (extras for onw) because if they are in the "top" code, then they can easily hide other variable rather than be 'local' in functions. So less warnings from PyCharm for one.